crates/sui-types/src/error.rs

```diff
@@ -669,6 +670,43 @@ pub enum SuiError {
     NoCertificateProvidedError,
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Weight(f64);
+
+impl Weight {
```

Comment:
now that this is moved out of the traffic control code it probably needs a more descriptive name and some comments